packages/toolpad-app/src/toolpad/DomLoader.tsx

```diff
@@ -75,8 +82,43 @@ export type DomAction =
       node: appDom.AppDomNode;
     };
 
+const undoStack: appDom.AppDom[] = [];
```

Comment:
Could we use just one stack and have some kind of position variable to define where we are in the stack?

Reply:
We probably could, though I don't seem much benefit

Reply:
it feels much more intuitive to me - unless it's much more difficult to implement?
That stack would kind of work like a history of the DOM.

Reply:
Ok, I've just tried to get it working. While I believe it's totally possible to do that, my initial impression is that it adds a context overhead when you have to think about "pointers" and resolve the values. Maybe it's subjective, but for me it's easier to think about 2 "buckets" - one for undo and another for redo, where redo is destroyed when new action is added to undo stack.
I'd prefer to keep this version and if you think this could be done elegantly with single stack and pointer, I'd be curious to see that version